floor/test/integration/view/view_test.dart

```diff
@@ -85,6 +104,48 @@ void main() {
         expect(actual, equals(expected));
       });
 
+      test('query view with all values as stream', () async {
+        final actual = nameDao.findAllNamesAsStream();
+        expect(
+            actual,
+            emitsInOrder(<List<Name>>[
+              [], //initial state
+              [Name('Frank'), Name('Leo')], // after inserting Persons
+              [
+                // after inserting Dog:
+                Name('Frank'),
+                Name('Leo'),
+                Name('Romeo')
+              ],
+              [
+                // after updating Leo:
+                Name('Frank'),
+                Name('Leonhard'),
+                Name('Romeo')
+              ],
+              [Name('Frank')], // after removing Person (and associated Dog)
+            ]));
+
+        final persons = [Person(1, 'Leo'), Person(2, 'Frank')];
+        await personDao.insertPersons(persons);
+
+        await simulateDoingSomethingElse();
```

Comment:
⚠️ `sleep()` also blocks asynchronous execution. This means that these tests only pass because sqflite spawns another Isolate under the hood? Dart FFI is still sort of unknown to me.
Introducing nondeterministic tests should only be considered as a last resort. Especially when introducing a test that takes 3 seconds. I played with it, trying to find a solution, without any luck. I guess it was the same for you, which made you introduce this function.
I would prefer using `Future.delayed()` instead of `sleep()`, which also seems to work with a delay of only 100 ms.

```suggestion
        await Future<void>.delayed(const Duration(milliseconds: 100));
```